client/src/details/SFSUrlBuilder.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "UrlBuilder.h"
+
+namespace SFS::details
+{
+// Wrapper struct that only exists to allow the SFSUrlBuilder to have a constructor that takes a custom URL
+struct SFSCustomUrl
+{
+    explicit SFSCustomUrl(std::string url) : url(std::move(url))
+    {
+    }
+
+    std::string url;
+};
+
+class SFSUrlBuilder : private UrlBuilder

```

Comment:
OOP nit: you have an IS-A relationship here but privately deriving from UrlBuilder.  Should UrlBuilder be a class member instead?

Reply:
No, most member methods are leveraged directly, this is just an extension of the other class

Reply:
I meant from a usage perspective not an outside perspective, e.g.

```
class Mammal {
  bool hasFur() {  }
 };
 
 class Human: private Mammal {
  ...
  };
  
  Human jeff = new Human();
  jeff.HasFur(); /* inaccessible? */
  ```